### PR TITLE
Topic search UI

### PR DESCRIPTION
**Pull request by finestructure**
Exposing keyword search in an initial version:

CleanShot 2021-06-28 at 08 40 35@2x

CleanShot 2021-06-28 at 08 40 52@2x

Labelling and naming the results to be tweaked but this adds the functionality.

### CODE REVIEW

File: Sources/App/Controllers/KeywordController.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+struct KeywordController {
+
+    static func query(on database: Database, keyword: String) -> EventLoopFuture<[Package]> {
+        Package.query(on: database)
+            .with(\.$repositories)
+            .join(Repository.self, on: \Repository.$package.$id == \Package.$id)
+            .filter(
+                DatabaseQuery.Field.path(Repository.path(for: \.$keywords), schema: Repository.schema),
+                DatabaseQuery.Filter.Method.custom("@>"),
+                DatabaseQuery.Value.bind([keyword])
```
Comment (finestructure) [inline Sources/App/Controllers/KeywordController.swift:15]: This is not the query I originally intended to write. Fluent doesn't seem to support expressing `'foo' LIKE ANY("keywords")` so this is an equivalent workaround which performs similarly.

I've to a question open in the Vapor Discord but not takers yet: https://discord.com/channels/431917998102675485/684159753189982218/858402370144960552

File: Sources/App/Core/SiteURL.swift [hunk 2]
```diff
@@ -196,6 +203,9 @@ enum SiteURL: Resourceable {
             case let .docs(next):
                 return ["docs"] + next.pathComponents
 
+            case .keywords:
+                return ["keywords", ":keyword"]
```
Comment (finestructure) [inline Sources/App/Core/SiteURL.swift:207]: FWIW, there's a "dead" `keywords` Github account which we're sort of overlaying here. But even so, `swiftpackageindex.com/keywords` should still work even if we add `swiftpackageindex.com/keywords/foo`.

CleanShot 2021-06-28 at 08 09 55@2x

**Comment by finestructure** [comment]
NB: #1198 which I'll address separately.

File: Sources/App/Core/SiteURL.swift (hunk 2 above)
Comment (daveverwer) [reply Sources/App/Core/SiteURL.swift:207]: Yea this is fine.

**Review by daveverwer — COMMENTED** [review]
I think we probably want to put the actual keyword searches behind a feature flag for now and have them only appear on staging. Couple of reasons:

* Keyword search makes less sense when the keywords are not exposed on the package pages, which Sarah is working on.
* I'd like to look at the display of search results and potentially bring icons (not emoji) into the list of search results.

If possible, we should only include keyword search results on staging deployments for now.

**Comment by finestructure** [comment]
Yeah, the UI is a placeholder, even more so for the upcoming author search where there's no applicable emoji 😅

I won't fully feature flag this but disable it for prod.

**Comment by finestructure** [comment]
This is good to go now.

**Review by daveverwer — APPROVED** [review]

**Comment by daveverwer** [comment]
All good to go. I'll let you rebase/merge in main and then merge the PR.

**Comment by finestructure** [comment]
BTW, those web view snapshots I included almost certainly need updating on Catalina, like the rest.

**Pull request merged**